ft/serialize/block_allocator.h

```diff
@@ -43,172 +43,155 @@ Copyright (c) 2006, 2015, Percona and/or its affiliates. All rights reserved.
 #include "portability/toku_pthread.h"
 #include "portability/toku_stdint.h"
 #include "portability/toku_stdlib.h"
-
 // Block allocator.
 //
 // A block allocator manages the allocation of variable-sized blocks.
 // The translation of block numbers to addresses is handled elsewhere.
 // The allocation of block numbers is handled elsewhere.
 //
 // When creating a block allocator we also specify a certain-sized
-// block at the beginning that is preallocated (and cannot be allocated or freed)
+// block at the beginning that is preallocated (and cannot be allocated or
+// freed)
 //
 // We can allocate blocks of a particular size at a particular location.
-// We can allocate blocks of a particular size at a location chosen by the allocator.
+// We can allocate blocks of a particular size at a location chosen by the
+// allocator.
 // We can free blocks.
 // We can determine the size of a block.
-
-class block_allocator {
-public:
+// This class is the base abstract class that defines the interface for
+// block_table
+class BlockAllocator {
+   public:
     static const size_t BLOCK_ALLOCATOR_ALIGNMENT = 4096;
 
     // How much must be reserved at the beginning for the block?
-    //  The actual header is 8+4+4+8+8_4+8+ the length of the db names + 1 pointer for each root.
+    //  The actual header is 8+4+4+8+8_4+8+ the length of the db names + 1
+    //  pointer for each root.
     //  So 4096 should be enough.
     static const size_t BLOCK_ALLOCATOR_HEADER_RESERVE = 4096;
-    
-    static_assert(BLOCK_ALLOCATOR_HEADER_RESERVE % BLOCK_ALLOCATOR_ALIGNMENT == 0,
+
+    static_assert(BLOCK_ALLOCATOR_HEADER_RESERVE % BLOCK_ALLOCATOR_ALIGNMENT ==
+                      0,
                   "block allocator header must have proper alignment");
 
-    static const size_t BLOCK_ALLOCATOR_TOTAL_HEADER_RESERVE = BLOCK_ALLOCATOR_HEADER_RESERVE * 2;
+    static const size_t BLOCK_ALLOCATOR_TOTAL_HEADER_RESERVE =
+        BLOCK_ALLOCATOR_HEADER_RESERVE * 2;
 
-    enum allocation_strategy {
+    struct BlockPair {
+        uint64_t _offset;
+        uint64_t _size;
+        BlockPair(uint64_t o, uint64_t s) : _offset(o), _size(s) {}
+        int operator<(const struct BlockPair &rhs) const {
+            return _offset < rhs._offset;
+        }
+        int operator<(const uint64_t &o) const { return _offset < o; }
+    };
+
+    enum EAllocationStrategy {
```

Comment:
I'm not sure that is what you think it is, it is a diff oddity, the actual code looks like:
```
    struct BlockPair {
        uint64_t _offset;
        uint64_t _size;
        BlockPair(uint64_t o, uint64_t s) : _offset(o), _size(s) {}
        int operator<(const struct BlockPair &rhs) const {
            return _offset < rhs._offset;
        }
        int operator<(const uint64_t &o) const { return _offset < o; }
    };

    enum EAllocationStrategy {
        BA_STRATEGY_FIRST_FIT = 1,
        BA_STRATEGY_BEST_FIT,
        BA_STRATEGY_PADDED_FIT,
        BA_STRATEGY_HEAT_ZONE
    };
```